### ISSUE TITLE
ROSCOE: What do premise and hypothesis mean in the human annotated datasets?

### ISSUE THREAD

**Issue by SahanaRamnath** [facebookresearch/ParlAI#4918]
@Golovneva 
I am training models on datasets other than the ones you have provided in the human annotated data (StrategyQA, to be specific). I have the predicted rationales and labels for each of my evaluation datapoints, and I want to obtain their ROSCOE scores using ```projects/roscoe/roscoe.py``` , but I am unsure what fields to use to describe these in the input json file.

The [ROSCOE paper](https://arxiv.org/pdf/2212.07919.pdf) talks about the source tokens (problem statement + question), hypothesis tokens (model's predicted reasoning steps + predicted answer), and optional reference tokens (gold reasoning steps). Meanwhile in function ```update_evaluator``` (line 83) in ```projects/roscoe/roscoe.py``` , as well as in the json files generated by ```roscoe_data/download_annotated.sh```, I see fields named ```premise```, ```hypothesis``` and ```gpt-3```. Could you clarify which of these corresponds to the tokens mentioned in the paper?

**Comment by Golovneva** [comment]
The main script to score chains is `projects/roscoe/score.py`. `projects/roscoe/roscoe.py` is a wrapper that does parsing and iterates over files. I would suggest two options to run it on external datasets:
1. (recommended) Adapt/Write your own version of `projects/roscoe/roscoe.py`, in particular:
- `def parse_chain` in 'ReasoningSteps' class should output a list of properly tokenized chain steps
- `def update_evaluator` in `ReasoningEvaluator` class is parsing each new input file and for each item extracts context string (problem+question), hypothesis string (reasoning + answer) and possibly a reference (if present). It's up to you to decide here which fields to use in each file
- main that iterates over files and sets which scores to run
2.  Create a script that will modify you file's format to be the same we used (see our data generated by `roscoe_data/download_annotated.sh` for the format). ESNLI, GSM8K, and three other datasets had slightly different formatting, you can follow any of those. In particular, there we already had context in "premise" field, sometimes followed by proposed answer (in "hypothesis" field), model-generated reasoning with answer in 'gpt-3', sometimes followed by the reference (different fields in different files). Parsing can be found in `def update_evaluator()` in `projects/roscoe/roscoe.py`

**Comment by Golovneva** [comment]
Glad to hear it worked!
Although we didn't find big difference between models we compared, on average using chain-finetuned model show higher correlations with final answer correctness, so we would recommend `model-name: facebook/roscoe-512-roberta-base` with 'model-type: sim_sce' as chain embedders.
As for `ppl-model-name`, any recent LLM should be fine, you can use default or anything more recent that is supported by HF in `AutoModelForCausalLM` class.

**Comment by SahanaRamnath** [comment]
Thank you!
